### DISCUSSION TITLE
Paste behavior of sudo statement sequence in ssh session

### DISCUSSION THREAD

**Discussion by marcfrei** [ghostty-org/ghostty#3068] [Q&A] [answered]
When pasting the following sequence of (passwordless) sudo commands into a ssh session, only the first command gets executed (clipboard-paste-protection = false):

```
sudo echo "A"
sudo echo "B"
```

Unexpected behavior with Ghostty for Mac (Version b92fb2d9, Build 8375, Commit b92fb2d9):

```
ubuntu@remote:~$ sudo echo "A"
sudo echo "B"A
ubuntu@remote:~$
```

Expected behavior with Terminal.app (Version 2.14 (455)):

```
ubuntu@remote:~$ sudo echo "A"
sudo echo "B"
A
B
ubuntu@remote:~$
```

Local host: macOS 15.2 (24C101)
Remote host: Ubuntu 24.04.1 LTS

**Comment by mitchellh** [comment]
I have to try to reproduce this but feel like it would’ve come up. Is this a terminfo issue?

**Comment by marcfrei** [comment]
Thanks, Mitchell.

I did some more testing and could reproduce the issue when connecting to two fresh EC2 instances, one running Amazon Linux 2023 and the other Ubuntu Server 24.04 LTS.

I'm not familiar with terminfo issues. How could I help debugging this?

**Reply by mitchellh** [reply, marked as answer]
I just removed the README section on it, but docs related to it will be available at this URL once the website ls launched in a couple days: http://ghostty.org/docs/help/terminfo You can also run the website locally today (the `website` repo) to view this.

**Reply by marcfrei** [reply]
Turns out this is indeed terminfo related. Copying over the terminfo entries or changing the local ssh config fixes the issue.

**Reply by marcfrei** [reply]
Thanks a lot and happy holidays!
